src/task.rs

```diff
@@ -53,14 +57,17 @@ impl Task {
             slots: Vec::new(),
             confirmed_start: None,
             confirmed_deadline: None,
-            internal_index: 0,
+            internal_marker,
         }
     }
 
     //TODO: The current way this is done may not be entirely accurate for tasks that can be done on
     //multiple days within certain time bounds.
     pub fn calculate_flexibility(&mut self) {
-        let hours_available = self.slots.len();
+        let mut hours_available = 0;
+        for slot in &self.slots {
+            hours_available += slot.num_hours();
+        }
         self.flexibility = hours_available - self.duration + 1;
```

Comment:
This assumes the Task is splittable into 1h portions that can be scheduled independently. Not always true.

Also, is it possible for this to become negative?

Reply:
Yes currently the code assumes this. Will need to add support for other scenarios.

Regards the negative flexibility, this could happen in the event that the calendar end cuts off the required number of hours to be assigned a task. For example if the calendar ends on a particular day at midnight, and a task `sleep` was to start at 22 and continue onto the next day.

So how I'm currently handling this, is the task's duration will be truncated (preventing the negative from occurring). In the output the sleep would go from 22 to midnight and get scheduled. I'm not sure if this is the best way to handle this and open to suggestions.

Reply:
Yeah the Task should have all the info to calculate its flexibility.

The sleep case is interesting.

For now I would prefer just adding a day to the calendar and truncating any Task that starts on the last day - rather than artificially reducing the duration. 

The idea of the Calendar is not that time 'ends' on that datetime (it's not the apocalypse) just rather the end of the requested time range on the calendar. Any Task that starts after the requested range can be discarded. Any Task with a due date before the end of the Calendar (view) must be possible.